### PR TITLE
Vnctl remove httparty

### PR DESCRIPTION
**Pull request by Metallion**
I miss programming a bit so I decided to quickly remove the httparty library from vnctl. Ruby has a perfectly usable HTTP class in the standard library and less dependencies is better.

### CODE REVIEW

File: vnctl/lib/vnctl/cli/base.rb [hunk 1]
```diff
@@ -189,27 +179,28 @@ def rel_name
             desc "add #{base_uuid_label} #{relation_uuid_label} OPTIONS",
               "Adds #{desc_label} to a(n) #{parent.namespace}."
             def add(base_uuid, rel_uuid)
-              puts post("#{suffix}/#{base_uuid}/#{rel_name}/#{rel_uuid}", :query => options)
+              full_uri_suffix = "#{suffix}/#{base_uuid}/#{rel_name}/#{rel_uuid}"
+              puts Vnctl.webapi.post(full_uri_suffix, options)
```
Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [DuplicateMethodCall] calls Vnctl.webapi 4 times

File: vnctl/lib/vnctl/cli/base.rb [hunk 2]
```diff
@@ -189,27 +179,28 @@ def rel_name
...
             end
           else
             desc "add #{base_uuid_label} OPTIONS",
               "Adds #{desc_label} to a(n) #{parent.namespace}."
             def add(base_uuid)
-              puts post("#{suffix}/#{base_uuid}/#{rel_name}", :query => options)
+              puts Vnctl.webapi.post("#{suffix}/#{base_uuid}/#{rel_name}", options)
```
Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [DuplicateMethodCall] calls Vnctl.webapi 4 times

File: vnctl/lib/vnctl/cli/base.rb [hunk 3]
```diff
@@ -189,27 +179,28 @@ def rel_name
...
             end
           end
 
           desc "show #{base_uuid_label}",
             "Shows all #{desc_label} in this #{parent.namespace}."
           def show(base_uuid)
-            puts get("#{suffix}/#{base_uuid}/#{rel_name}")
+            puts Vnctl.webapi.get("#{suffix}/#{base_uuid}/#{rel_name}")
```
Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [DuplicateMethodCall] calls Vnctl.webapi 4 times

File: vnctl/lib/vnctl/cli/base.rb [hunk 4]
```diff
@@ -189,27 +179,28 @@ def rel_name
...
           end
 
           desc "del #{base_uuid_label} #{relation_uuid_label}(S)",
             "Removes #{desc_label} from a(n) #{parent.namespace}."
           def del(base_uuid, *rel_uuids)
             puts rel_uuids.map { |rel_uuid|
-              delete("#{suffix}/#{base_uuid}/#{rel_name}/#{rel_uuid}")
+              Vnctl.webapi.delete("#{suffix}/#{base_uuid}/#{rel_name}/#{rel_uuid}")
```
Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [DuplicateMethodCall] calls Vnctl.webapi 4 times

File: vnctl/lib/vnctl/cli/base.rb [hunk 5]
```diff
@@ -94,9 +84,9 @@ def self.define_show
         desc "show [UUID(S)]", "Shows all or a specific set of #{namespace}(s)."
         define_method(:show) do |*uuids|
           if uuids.empty?
-            puts get(suffix)
+            puts Vnctl.webapi.get(suffix)
```
Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [DuplicateMethodCall] calls Vnctl.webapi 2 times

File: vnctl/lib/vnctl/cli/base.rb [hunk 6]
```diff
@@ -94,9 +84,9 @@ def self.define_show
...
           else
-            uuids.each { |uuid| puts get("#{suffix}/#{uuid}") }
+            uuids.each { |uuid| puts Vnctl.webapi.get("#{suffix}/#{uuid}") }
```
Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [DuplicateMethodCall] calls Vnctl.webapi 2 times

Comment (hansode) [inline vnctl/lib/vnctl/cli/base.rb]: [reek] [NestedIterators] contains iterators nested 2 deep

File: vnctl/lib/vnctl/webapi.rb [hunk 7]
```diff
@@ -2,7 +2,44 @@
...
+      @webapi_port = Vnctl.conf.webapi_port.to_i
+      @webapi_version = Vnctl.conf.webapi_version
+      @webapi_protocol = Vnctl.conf.webapi_protocol
+      @output_format = Vnctl.conf.output_format
+
+      @webapi_full_uri = "%s://%s:%s/api/%s" %
+        [@webapi_protocol, @webapi_uri, @webapi_port, @webapi_version]
+    end
+
+    def send_request(verb, suffix, params = nil)
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DataClump] takes parameters [params, suffix] to 5 methods

File: vnctl/lib/vnctl/webapi.rb [hunk 8]
```diff
@@ -2,7 +2,44 @@
...
+
+      response = Net::HTTP.start(uri.host, uri.port) do |http|
+        request = verb.new(uri.request_uri)
+        http.request(request)
+      end
+
+      response.body
+    end
+
+    def post(suffix, params = nil)
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DataClump] takes parameters [params, suffix] to 5 methods

File: vnctl/lib/vnctl/webapi.rb [hunk 9]
```diff
@@ -2,7 +2,44 @@
...
+      send_request(Net::HTTP::Post, suffix, params)
+    end
+
+    def get(suffix, params = nil)
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DataClump] takes parameters [params, suffix] to 5 methods

File: vnctl/lib/vnctl/webapi.rb [hunk 10]
```diff
@@ -2,7 +2,44 @@
...
+      send_request(Net::HTTP::Get, suffix, params)
+    end
+
+    def put(suffix, params = nil)
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DataClump] takes parameters [params, suffix] to 5 methods

File: vnctl/lib/vnctl/webapi.rb [hunk 11]
```diff
@@ -2,7 +2,44 @@
...
+      send_request(Net::HTTP::Put, suffix, params)
+    end
+
+    def delete(suffix, params = nil)
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DataClump] takes parameters [params, suffix] to 5 methods

File: vnctl/lib/vnctl/webapi.rb [hunk 12]
```diff
@@ -2,7 +2,44 @@
 
 module Vnctl
   class WebApi
-    include HTTParty
-    base_uri "#{Vnctl.conf.webapi_uri}:#{Vnctl.conf.webapi_port}"
+    def initialize
+      @webapi_uri = Vnctl.conf.webapi_uri
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DuplicateMethodCall] calls Vnctl.conf 5 times

File: vnctl/lib/vnctl/webapi.rb [hunk 13]
```diff
@@ -2,7 +2,44 @@
...
+      @webapi_port = Vnctl.conf.webapi_port.to_i
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DuplicateMethodCall] calls Vnctl.conf 5 times

File: vnctl/lib/vnctl/webapi.rb [hunk 14]
```diff
@@ -2,7 +2,44 @@
...
+      @webapi_version = Vnctl.conf.webapi_version
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DuplicateMethodCall] calls Vnctl.conf 5 times

File: vnctl/lib/vnctl/webapi.rb [hunk 15]
```diff
@@ -2,7 +2,44 @@
...
+      @webapi_protocol = Vnctl.conf.webapi_protocol
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DuplicateMethodCall] calls Vnctl.conf 5 times

File: vnctl/lib/vnctl/webapi.rb [hunk 16]
```diff
@@ -2,7 +2,44 @@
...
+      @output_format = Vnctl.conf.output_format
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DuplicateMethodCall] calls Vnctl.conf 5 times

File: vnctl/lib/vnctl/webapi.rb (hunk 7 above)
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [TooManyStatements] has approx 6 statements

Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [FeatureEnvy] refers to uri more than self

File: vnctl/lib/vnctl/webapi.rb [hunk 17]
```diff
@@ -2,7 +2,44 @@
...
+
+      @output_format = @conf.output_format
+
+      @webapi_full_uri = "%s://%s:%s/api/%s" % [@conf.webapi_protocol,
+                                                @conf.webapi_uri,
+                                                @conf.webapi_port,
+                                                @conf.webapi_version]
+    end
+
+    def send_request(verb, suffix, params = nil)
```
Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [DataClump] takes parameters [params, suffix] to 5 methods

Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [TooManyStatements] has approx 6 statements

Comment (hansode) [inline vnctl/lib/vnctl/webapi.rb]: [reek] [FeatureEnvy] refers to uri more than self

File: vnctl/lib/vnctl/cli/base.rb (hunk 1 above)
Comment (Metallion) [inline vnctl/lib/vnctl/cli/base.rb]: I think this reek is fine. `Vnctl.webapi` accesses a global variable holding a WebAPI object than can be used for sending HTTP requests.

File: vnctl/lib/vnctl/webapi.rb (hunk 8 above)
Comment (Metallion) [inline vnctl/lib/vnctl/webapi.rb]: This reek is also fine. There is one main method and then 4 shortcuts that fill in the `verb` parameter.

File: vnctl/lib/vnctl/webapi.rb (hunk 17 above)
Comment (Metallion) [inline vnctl/lib/vnctl/webapi.rb]: > [TooManyStatements]

They're all statements completing the single simple job of making an HTTP request and returning the response body.

> [FeatureEnvy]

`uri` is a local variable only used in this particular requests. These references are fine.

File: vnctl/lib/vnctl/cli/base.rb (hunk 6 above)
Comment (Metallion) [inline vnctl/lib/vnctl/cli/base.rb]: I don't see nested iterators. Perhaps the bot thinks `define_method(:show) do |*uuids|` is also an iteration?

**Comment by akry** [comment]
+1

**Pull request merged**